### PR TITLE
Fix appendTitle, prependTitle. Update unit test for append, prepend and setTitle.

### PR DESCRIPTION
**Pull request by borisdelev**
Update tests for prependTitle and appendTitle functions.
For details:  #12233 

Update test for setTitle and getTitle: Make sure that getTitle(false) did not return <title> tag element

### CODE REVIEW

**Comment by butamuh4o** [comment]
Hope i make it right now. Sorry for my mistakes :/

**Review by sergeyklay — APPROVED** [review]

**Comment by virgofx** [comment]
Unit tests look fine ... can we get can update though for the actual Tag implementation so this will pass?

**Comment by butamuh4o** [comment]
@virgofx i will try to make this changes too...

**Comment by sergeyklay** [comment]
@butamuh4o As you can see tests fails

**Comment by butamuh4o** [comment]
@sergeyklay I want to ask you to help me with something about a change i was trying to do. Can u check that and give me some details where and what is wrong with syntax? Thank you!!!

```
	/**
	 * Prepends a text to current document title
	 */
	public static function prependTitle(var title) -> void
	{
		var prepend = [];
		if self::_documentPrependTitle !== null {
			let prepend = self::_documentPrependTitle ;
		}
		
		if typeof title == array {
			let prepend = title;
		} else {
			if typeof title == string {
				array_unshift(prepend, title) ;
			}
		}
		
		if !empty prepend {
			let self::_documentPrependTitle = prepend ;
		}
	}
```

File: phalcon/tag.zep [hunk 1]
```diff
@@ -1061,17 +1061,51 @@ class Tag
...
-	public static function prependTitle(string title) -> void
+	public static function prependTitle(var title) -> void
 	{
-		let self::_documentPrependTitle = title;
+		var prepend = [];
+		if self::_documentPrependTitle !== null {
+			let prepend = self::_documentPrependTitle ;
+		}
+		
+		if typeof title == "array" {
```
Comment (sergeyklay) [inline phalcon/tag.zep]: This is new feature request?

File: phalcon/tag.zep [hunk 2]
```diff
@@ -1061,17 +1061,51 @@ class Tag
...
+			if typeof title == "string" {
+				array_unshift(prepend, title) ;
+			}
+		}
+		
+		if empty prepend {
+			let prepend = null ;
+		}
+		
+		let self::_documentPrependTitle = prepend ;
```
Comment (sergeyklay) [inline phalcon/tag.zep]: So now `_documentPrependTitle` is array or null and this breaking BC

**Review by sergeyklay — CHANGES_REQUESTED** [review]

**Comment by butamuh4o** [comment]
No it is not change request. Just think it is better to have that option. Will revert that idea. Sorry

**Comment by sergeyklay** [comment]
It is good idea, but for 3.1.x not for 3.0.x

**Comment by sergeyklay** [comment]
@butamuh4o Also could you update CHANGELOG.md?

**Comment by butamuh4o** [comment]
So i will make a clear pull request and update changelog. Just to get to the office. Thank you!

**Pull request closed without merging**